scubagoggles/rego/Drive.rego

```diff
@@ -0,0 +1,1588 @@
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs2_3,
+        "NonCompliantGroups": NonCompliantGroups2_3},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check2_3_OK
+    Conditions := {count(NonCompliantOUs2_3) == 0, count(NonCompliantGroups2_3) == 0 }
+    Status := (false in Conditions) == false
+}
+#--
+
+#
+# Baseline GWS.DRIVEDOCS.2.4
+#--
+
+DriveId2_4 := utils.PolicyIdWithSuffix("GWS.DRIVEDOCS.2.4")
+
+LogMessage2_4 := "Shared Drive Creation new_team_drive_restricts_download"
+
+Check2_4_OK if {
+    not PolicyApiInUse
+    events := utils.FilterEventsOU(LogEvents, LogMessage2_4, utils.TopLevelOU)
+    count(events) > 0
+}
+
+Check2_4_OK if {PolicyApiInUse}
+
+NonComplianceMessage2_4 := "Viewers and commenters are allowed to download, print, and copy files."
+
+NonCompliantOUs2_4 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage2_4
+}
+if {
+    not PolicyApiInUse
+    some OU in utils.OUsWithEvents
+    Events := utils.FilterEventsOU(LogEvents, LogMessage2_4, OU)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    contains("false", LastEvent.NewValue) == true
+    LastEvent.NewValue != "DELETE_APPLICATION_SETTING"
+}
+
+NonCompliantGroups2_4 contains {
+    "Name": Group,
+    "Value": NonComplianceMessage2_4
+}
+if {
+    not PolicyApiInUse
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, LogMessage2_4, Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    contains("false", LastEvent.NewValue) == true
+    LastEvent.NewValue != "DELETE_APPLICATION_SETTING"
+}
+
+NonCompliantOUs2_4 contains {
+    "Name": OU,
+    "Value": NonComplianceMessage2_4
+}
+if {
+    some OU, settings in input.policies
+    DriveEnabled(OU)
+    allowPrint := settings.drive_and_docs_shared_drive_creation.allowedPartiesForDownloadPrintCopy
+    allowPrint != "EDITORS_ONLY"
+}
+
+tests contains {
+    "PolicyId": DriveId2_4,
+    "Criticality": "Shall",
+    "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
+    "ActualValue": "No relevant event for the top-level OU in the current logs",
+    "RequirementMet": DefaultSafe,
+    "NoSuchEvent": true
+}
+if {
+    not PolicyApiInUse
+    DefaultSafe := true
+    not Check2_4_OK
+}
+
+tests contains {
+    "PolicyId": DriveId2_4,
+    "Criticality": "Shall",
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs2_4, NonCompliantGroups2_4),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs2_4,
+        "NonCompliantGroups": NonCompliantGroups2_4},
+    "RequirementMet": Status,
+    "NoSuchEvent": false
+}
+if {
+    Check2_4_OK
+    Conditions := {count(NonCompliantOUs2_4) == 0, count(NonCompliantGroups2_4) == 0 }
+    Status := (false in Conditions) == false
+}
+#--
+
+###################
+# GWS.DRIVEDOCS.3 #
+###################
+
+#
+# Baseline GWS.DRIVEDOCS.3.1
+#--
```

Comment:
The report fails you no matter what you select for the security update settings

Reply:
This one doesn't have a flag about fixed settings due to the free license, but I suspect this is the case.  The setting for "
Allow users to remove/apply the security update for files they own or manage" is True no matter the state of the checkbox.

![image](https://github.com/user-attachments/assets/d14a5649-214d-44bf-a67b-4b6933b8016b)

`'drive_and_docs_file_security_update': {'allowUsersToManageUpdate': True,`

I will check that this might be a fixed setting at the Monday Google meeting.

Reply:
After implementing Google's policy "reduction" process, this is working for me
when I change settings in the Admin UI.